### ISSUE TITLE
User SBAT sign transactions and access valora user info from the browser

### ISSUE THREAD

**Issue by critesjosh** [celo-org/celo-monorepo#6657]
### Expected Behavior

After approving an action (signing in or signing a transaction) in Valora, the user should be redirected to the browser page that they requested the action from. 

### Current Behavior

_I'm trying to sign transaction or access valora user info from web brower. And it works fine in mobile safari. Valora app opened, then I confirm transaction and get redirection back to the same tab of Safari.
But in Chrome it works differently. After I approve transaction in Valora app, I always redirected to new tab in Chrome. So it's completely wrong._

_I use import `{ requestTxSig } from '@celo/dappkit'`  which uses inside `import { Linking } from 'expo'` and Linking.openURL(...)_

_I tried both WebBrowser and Linking modules._

[source (Discord)](https://discord.com/channels/600834479145353243/607994529311883274/791936250898415620)

**Comment by critesjosh** [comment]
A solution could be to create a tutorial about how to do this the right way

**Comment by barbaraliau** [comment]
Related to version upgrades?

**Comment by critesjosh** [comment]
here is the github repo with the application that is running into problems https://github.com/Ambisafe/moola-app

**Comment by critesjosh** [comment]
this should be unblocked now, I just have the savings circle dapp example left to update

**Comment by nixoid** [comment]
@critesjosh it's not clear from your comment whether you guys have fixed anything in the framework or what do you mean by "unblocked". If you can post some working code example here I would appreciate

**Comment by jsdmc** [comment]
Here is example of deeplink request that is causing problems
`celo://wallet/dappkit?type=sign_tx&requestId=deposit&callback=http%3A%2F%2Flocalhost%3A3000%2F%23%2Fdeposit%2FCELO%2F0.07466996135339098215&dappName=Moola&txs=W3sidHhEYXRhIjoiMHhkMmQwZTA2NjAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMGVlZWVlZWVlZWVlZWVlZWVlZWVlZWVlZWVlZWVlZWVlZWVlZWVlZWUwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMTA5NDdlZjI5ZGM2Yjg2MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMCIsImVzdGltYXRlZEdhcyI6MjAwMDAwMCwiZnJvbSI6IjB4MDdlZjI5MGM5Y2M2MDIxOTFiNmQ4MDY4OThhMjFmNjA4NTcxMzQzMiIsInRvIjoiMHhjMTU0OEY1QUExRDc2Q0RjQUI3Mzg1RkE2QjVjRUE3MGY5NDFlNTM1Iiwibm9uY2UiOjI3LCJ2YWx1ZSI6Ijc0NjY5OTYxMzUzMzkwOTgyIn1d`

**Comment by jsdmc** [comment]
And that's error I can take from logs
```
Error: Transaction has been reverted by the EVM:
{
  "blockHash": "0xbac27a1a68d7922c4819881748b9650a797d2704feb30a8fc322ae3f005e6d6c",
  "blockNumber": 5045880,
  "contractAddress": null,
  "cumulativeGasUsed": 328720,
  "from": "0x07ef290c9cc602191b6d806898a21f6085713432",
  "gasUsed": 173960,
  "logs": [],
  "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "status": false,
  "to": "0xc1548f5aa1d76cdcab7385fa6b5cea70f941e535",
  "transactionHash": "0x06916a3bd0ecc997e58450f4770a73deee41279a7114eb167ca4e7675045650e",
  "transactionIndex": 1
}
```

**Comment by jsdmc** [comment]
I guess a problem is somewhere around here
https://github.com/celo-org/celo-monorepo/blob/fce3df17d1df43c37e59a3cc6918cc1a788a7dad/packages/sdk/utils/src/dappkit.ts#L262

**Comment by critesjosh** [comment]
@nixoid, this issue was marked as blocked by another issue that I was working on. I was just saying that I finished work on the other issue, so this issue is ready to be worked on.

@jsdmc thanks for sharing this info. From what I understand, this issue is about the UI link between Valora and a web app, which works in Safari but not in Chrome. Does the transaction revert only when using Chrome? Also, just curious if you know why the transaction you shared is failing?

**Comment by patricknbaron** [comment]
The transaction reverts in Chrome, Firefox, and Brave.

**Comment by jsdmc** [comment]
Guys. I'm sorry, was in rush and completely missed that it's a different issue. 
Everything I wrote above is related to this issue
https://github.com/celo-org/celo-monorepo/issues/6830

I posted details it appropriate issue.

**Comment by eelanagaraj** [comment]
No worries @jsdmc!

Additional context re: tab-switching bug from our chat (for context here):
- Run in expo web mode in mobile browser
- Seems to be a problem on Android in particular, should also be observable on app.moola.market
- Works on iPhone (tested on XR) with Chrome, Brave, Safari (login through Celo Alfajores wallet). Note that for both Chrome/Brave, after authentication in wallet, user is logged into moola but in a separate tab than original (i.e. creates a new tab), but they are logged in properly

Logic flow:
- user in Moola app clicks deeplink "Connect Valora" in a browser on their mobile device
- user in same tab is redirected to the Valora app on their mobile device to sign prepopulated TX
- once the TX is signed, user should be redirected back to Moola web app in the browser on their device
- in Chrome, the user is instead redirected to a new (blank) tab; in Safari and Chrome/Brave on iPhone (tested on XR) user is properly directed back to the Moola app

**Comment by critesjosh** [comment]
@patricknbaron let me know that Valora v1.9.2 resolved the deep linking to chrome issue.

**update**: this issue is not resolved

**Comment by barbaraliau** [comment]
@jsdmc, @patricknbaron, if this is still an issue, please feel free to reopen with additional details.
